portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,13 +37,16 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-
+    String stringNumberOfMessages = request.getParameter("nr");
```

Comment:
Consider using request.getParameter("nr") directly in the line below instead of defining this variable.

Reply:
Solved